patches/tModLoader/Terraria/ModLoader/ModContent.cs

```diff
@@ -429,6 +429,7 @@ internal static bool TryGetTexture(string name, out Asset<Texture2D> texture)
 		}
 
 		internal static void UnloadModContent() {
+			MenuLoader.currentModMenu = null; // This is necessary to do before anything else because the main menu drawing is on a separate thread and it can cause issues with trying to use disposed textures
```

Comment:
~~Seems like it could work, but you need a better comment here and where you check `currentModMenu` for null. Noting that it's _only_ null during mod reloading, and that none of the ModMenu code runs during reloading, it's all just back to vanilla~~

Letting `currentMenu` be null seems like a recipe for disaster. I think the vanilla and tML menus should just be always active, and this code should just switch back to the tML menu _if_ they were on a mod one

Reply:
there's a check in the Main patch for the menu being null - it just resorts to vanilla drawcode if that's the case. I've checked and it should all be null safe